core/storage/src/storages/beacon.rs

```diff
@@ -1,9 +1,19 @@
-use super::{BlockChainStorage, GenericStorage};
+use super::{
+    BlockChainStorage, GenericStorage, COL_PROPOSAL, 
+    COL_PARTICIPATION, COL_THRESHOLD, COL_ACCEPTED_AUTHORITY,
+    COL_PROCESSED_BLOCKS
+};
 use crate::storages::ChainId;
 use crate::KeyValueDB;
-use primitives::beacon::SignedBeaconBlock;
-use primitives::beacon::SignedBeaconBlockHeader;
+use primitives::beacon::{SignedBeaconBlock, SignedBeaconBlockHeader};
+use primitives::types::{AuthorityStake, AuthorityMask};
+use primitives::serialize::{Encode, Decode};
 use std::sync::Arc;
+use std::collections::{HashMap, HashSet};
+use std::hash::Hash;
+
+type Epoch = u64;
```

Comment:
`Epoch` and `Slot` types are already defined somewhere else, see authority. Use `use` instead of redefining them.

Reply:
It was defined in authority. To avoid cyclic dependency, I moved them to primitives.